client/src/cmdhfemrtd.c

```diff
@@ -711,18 +710,17 @@ static bool emrtd_select_and_read(uint8_t *dataout, int *dataoutlen, const char
 }
 
 static bool emrtd_dump_ef_dg2(uint8_t *file_contents, int file_length) {
-    uint8_t data[EMRTD_MAX_FILE_SIZE];
-    int datalen = 0;
+    int offset, datalen = 0;
 
     // This is a hacky impl that just looks for the image header. I'll improve it eventually.
     // based on mrpkey.py
     // FF D8 FF E0 -> JPEG
     // 00 00 00 0C 6A 50 -> JPEG 2000
-    for (int i = 0; i < file_length - 6; i++) {
-        if ((file_contents[i] == 0xFF && file_contents[i + 1] == 0xD8 && file_contents[i + 2] == 0xFF && file_contents[i + 3] == 0xE0) ||
-            (file_contents[i] == 0x00 && file_contents[i + 1] == 0x00 && file_contents[i + 2] == 0x00 && file_contents[i + 3] == 0x0C && file_contents[i + 4] == 0x6A && file_contents[i + 5] == 0x50)) {
-            datalen = file_length - i;
-            memcpy(data, file_contents + i, datalen);
+    // Note: Doing file_length - 6 to account for the longest data we're checking.
+    for (offset = 0; offset < file_length - 6; offset++) {
+        if ((file_contents[offset] == 0xFF && file_contents[offset + 1] == 0xD8 && file_contents[offset + 2] == 0xFF && file_contents[offset + 3] == 0xE0) ||
+            (file_contents[offset] == 0x00 && file_contents[offset + 1] == 0x00 && file_contents[offset + 2] == 0x00 && file_contents[offset + 3] == 0x0C && file_contents[offset + 4] == 0x6A && file_contents[offset + 5] == 0x50)) {
+            datalen = file_length - offset;
```

Comment:
Use memcmp  fct ?

Reply:
Considered that, would likely have a slightly higher overhead, but I'll still do it for better visibility later today.